nomad/local/start_development_environment.sh

```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+if [[ $EUID -ne 0 ]]; then
+    echo "This script must be run as root"
+    exit 1
+fi
```

Comment:
It may be useful to put a comment here that this guard is strictly informative; `nomad agent -dev-connect` _can't_ run as a non-root user.

(That is to say, some future engineer working on this script isn't going to horribly break things if they happen to remove this check. Also, Nomad outputs an error message saying it has to run as root anyway.)